apps/campus/models.py

```diff
@@ -231,11 +231,20 @@ def flat(l):
 
     def _link(self):
         url = reverse('location', kwargs={'loc': self.id})
-        return '<a href="%s%s/" data-pk="%s">%s</a>' % (url, slugify(self.name), self.id, self.title)
+
+        if self.object_type in settings.REDIRECT_TYPES:
+            url = "{0}{1}".format(settings.LOCATION_REDIRECT_BASE, slugify(self.name))
```

Comment:
It might be worth adding a separate function that performs the `"{0}{1}".format(settings.LOCATION_REDIRECT_BASE, slugify(location.name))` formatting, just for DRY-ness